### PR TITLE
Add the Suggest field to the SearchResp struct

### PR DESCRIPTION
**Pull request by zr-bennettm**
### Description
Adding the `Explanation` field to the 

### Issues Resolved
Closes issue [#601](https://github.com/opensearch-project/opensearch-go/issues/601) 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
Thanks for this! I fixed integration tests in #600, FYI.

We also have https://github.com/opensearch-project/opensearch-go/issues/284 open. Would you mind please checking whether https://github.com/opensearch-project/opensearch-api-specification has this field and a test for it in https://github.com/opensearch-project/opensearch-api-specification/tree/main/tests, and if not maybe adding it? We want to make sure we don't introduce regressions when we replace this code with generated one (can use help on that too ;)).

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Bump `github.com/aws/aws-sdk-go` from 1.54.12 to 1.55.5 ([#583](https://github.com/opensearch-project/opensearch-go/pull/583), [#590](https://github.com/opensearch-project/opensearch-go/pull/590), [#595](https://github.com/opensearch-project/opensearch-go/pull/595), [#596](https://github.com/opensearch-project/opensearch-go/pull/596))
 
 ### Added
+- Adds the `Suggest` field to the `SearchResp` struct. ([#602](https://github.com/opensearch-project/opensearch-go/pull/602))
```
Comment (dblock) [inline CHANGELOG.md:12]:
```suggestion
- Adds `Suggest` to `SearchResp` ([#602](https://github.com/opensearch-project/opensearch-go/pull/602))
```

(nitpicking on periods and extra words ;))

**Review by dblock — APPROVED** [review]
See nit below and let's wait for integ tests to go back to green and rebase.

**Review by dblock — APPROVED** [review]

**Comment by zr-bennettm** [comment]
thanks for the quick approval @dblock ! Could I expect to see this be part of the next release (4.2)?

**Comment by dblock** [comment]
For sure. If you really, really need a release soon(er) open a "Release v. Next" issue and we'll take care of it.

**Pull request merged**